src/net/sourceforge/kolmafia/textui/command/DemonNamesCommand.java

```diff
@@ -1,16 +1,54 @@
 package net.sourceforge.kolmafia.textui.command;
 
+import java.util.Arrays;
+import java.util.stream.Collectors;
 import net.sourceforge.kolmafia.KoLAdventure;
 import net.sourceforge.kolmafia.RequestLogger;
 import net.sourceforge.kolmafia.preferences.Preferences;
+import net.sourceforge.kolmafia.session.DemonName14Manager;
 
 public class DemonNamesCommand extends AbstractCommand {
   {
-    this.usage = " - list the demon names you know.";
+    this.usage =
+        "|solve14. With no param, list the demon names you know. With solve14, attempt to solve the 14th demon.";
   }
 
   @Override
   public void run(final String cmd, final String parameters) {
+    if (parameters.trim().equals("solve14")) {
+      var prefValue = Preferences.getString("demonName14Segments");
+
+      if (prefValue.isEmpty()) {
+        RequestLogger.printLine(
+            "You need to make bad requests with your Allied Radio Backpack to find segments of your demon name");
+        return;
+      }
+
+      var segments = Arrays.stream(prefValue.split(",")).collect(Collectors.toSet());
+
+      var solutions = DemonName14Manager.solve(segments);
```

Comment:
a statement like "this may take a while" would not go amiss, e.g. Just Eyes' Cut,Car,Kul,Cul,Bal,utK,rCa,rGu,arG,lBa,ilB,tKi,lCu,lCu,ulK,tKi takes 30 seconds to return 1011 results, so it's nice to let people know Mafia hasn't frozen